lib/nx_deflate.c

```diff
@@ -892,8 +892,12 @@ static void nx_compress_block_get_cpb(nx_streamp s, int fc)
    return.
 */
 
-/* updates stream offsets and also sets the block final bit */
-static int  nx_compress_block_update_offsets(nx_streamp s, int fc)
+/*
+   updates stream offsets and also sets the block final bit.
+   Inlining is avoided here because of a bug on GCC 10 that makes one of our
+   tests fail under valgrind.
+*/
+static int __attribute__((noinline)) nx_compress_block_update_offsets(nx_streamp s, int fc)
```

Comment:
Considering that we can reproduce this issue only with GCC 10, can we stop inlining only in that case?
Our new configure may help you with this.

Reply:
Done.